adrs/2068541249/README.md

```diff
@@ -0,0 +1,154 @@
+│   └── http-client.ts
+```
+
+There are some caveats configuring such files.
+
+For example, using the following configuration:
+
+```json
+{
+  "compilerOptions": {
+    "baseUrl": "src"
+  }
+}
+```
+
+That means, you could import the code as if those files or directories were
+modules:
+
+```tsx
+import { httpClient } from 'http-client';
+import { useMyQuery } from 'react-query';
+```
+
+That configuration brings some critical questions about the application.
+
+How do we know if those imports come from `node_modules` or the imports are
+part of our codebase?
+
+Although it may be simply to answer the question by scanning the directories and
+files under `./src`.
+
+In other cases like `import { useMyQuery } from 'react-query';` where you
+may have `react-query` as a dependency, and you used `./src/react-query`
+directory to store your React Query related things as an example, you need to
+scan the files to make sure that the imports don't belong to a dependency
+package.
+
+Furthermore, how does the module resolve resolution works?
+
+It may not be possible to answer such a question without knowing the underline
+tools configurations.
+
+It may be the case that a tool would scan `./src` first, and then `node_modules`,
+or vice-versa.
+
+That means that you may be overwriting dependency packages in cases that you
+use the same name of a dependency package as name of a directory allocated in
+`./src`, introducing potential bugs that are hard to track without knowing
+the underline tool.
+
+You may also run into such issues by changing `compilerOptions.paths` instead,
+for example:
+
+```jsonc
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "design/*": ["./src/design/*"]
+    }
+  }
+}
+```
+
+```tsx
+import { goldenRatio } from 'design/golder-ratio';
+```
+
+Same questions,
+
+* How do we know if those imports come from `node_modules` or it is part of our
+  codebase?
+* How does the module resolve resolution works in cases when somebody installs
+  `design` package as a dependency?
+
+The underline problem is registering a valid NPM package name as an alias,
+that introduce cognitive load, and potentially becoming more error-prune,
+relying on engineer maturity to understand how to debug, and outcome the
+potential issues.
+
+As the application grows, such practice makes the development harder, and also
+make the tools slower since it will require to scan `node_modules` for those
+aliases as well since it may be the case that those modules are a dependency
+module.
+
+## Resolution
+
+Since `@/` is not a valid NPM package, it is popular among other projects as
+well.
+
+* You **must** never change `baseUrl` from the root of the project.
+* You **must** use the following configuration to resolve files relative to the
+  root of the project:
+
+    ```jsonc
+    {
+      "compilerOptions": {
+        "baseUrl": ".", // The baseUrl is at the root of the project
+        "paths": {
+          "@/*": ["./src/*"]
```

Comment:
I actually ran into an issue with this recently. I can't remember it off the top of my head. I'll see if I can dig it up.

Reply:
https://github.com/facebook/create-react-app/issues/9406

Reply:
Following the issue, I changed the ADR to be more generic and agnostic of `{ts,js}config.json` but to be more focused on the issue and guidelines about it.